File: sample/src/main/java/io/snabble/testapp/PaymentCredentialsListActivity.java
package io.snabble.testapp;

import androidx.fragment.app.Fragment;

import io.snabble.sdk.ui.integration.PaymentCredentialsListFragment;

public class PaymentCredentialsListActivity extends BaseActivity{
    @Override
    public Fragment onCreateFragment() {
        Fragment fragment = new PaymentCredentialsListFragment();
        fragment.setArguments(getIntent().getBundleExtra("args"));
        return fragment;
    }
}

File: ui/src/main/java/io/snabble/sdk/ui/payment/PaymentCredentialsListActivity.kt
package io.snabble.sdk.ui.payment

import androidx.fragment.app.Fragment
import io.snabble.sdk.ui.BaseFragmentActivity

class PaymentCredentialsListActivity : BaseFragmentActivity() {
    companion object {
        const val ARG_PAYMENT_TYPE = PaymentCredentialsListView.ARG_PAYMENT_TYPE
        const val ARG_PROJECT_ID = PaymentCredentialsListView.ARG_PROJECT_ID
    }

    override fun onCreateFragment(): Fragment {
        val fragment = PaymentCredentialsListFragment()
        fragment.arguments = intent.extras
        return fragment
    }
}